volumina/pixelpipeline/datasources/interface.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+from typing import Optional
+from abc import ABC, abstractproperty, abstractmethod
+
+import numpy as np
+
+from volumina.utility.qabc import QABC, abstractsignal
+
+
+__all__ = ["DataSourceABC", "DataRequestABC", "IndeterminateRequestError"]
+
+
+class IndeterminateRequestError(Exception):
+    """
+    Raised if a request cannot be created or cannot be executed
+    because its underlying datasource is in an indeterminate state.
+    In such cases, the requester should simply ignore the error.
+    The datasource has the responsibility of sending a dirty notification
+    when the source is ready again.
+    """
+
+    pass
+
+
+class DataRequestABC(ABC):
```

Comment:
Is there a good reason why this does not reside in `pixelpipeline.asyncabcs`?

Reply:
Some requests (e.g. GrayscaleImageRequest) don't implement getResult method.

Reply:
sure, I mean the whole interface - i'd say it qualifies as async?

Reply:
I am not even sure that `asyncabc` is a reasonable name :) It should be also `interfaces.py` or something